src/charm.py

```diff
@@ -633,7 +669,66 @@ def _get_backup_key(self) -> str:
         timestamp = datetime.datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
         return f"{BACKUP_KEY_PREFIX}-{self.model.name}-{timestamp}"
 
-    def _get_ca_cert_location_in_charm(self) -> str:
+    def _on_new_access_certificate_joined(self, event: RelationJoinedEvent):
+        if not self._ingress_address:
+            raise RuntimeError("Tried to process relation join before peer relation.")
+        if not (private_key := self._get_private_key_from_workload().encode()):
+            private_key = generate_private_key()
+        new_csr = generate_csr(private_key=private_key, subject=self._ingress_address)
+        self._push_csr_to_workload(new_csr.decode())
+        self._push_unit_pk_to_workload(private_key=private_key.decode())
+        self.tls_certificate_access.request_certificate_creation(
+            certificate_signing_request=new_csr
+        )
+        logger.info(
+            "New certificate requested from %s for unit %s",
+            event.relation.name,
+            self.unit.name,
+        )
+
+    def _on_new_access_certificate_available(self, event: CertificateAvailableEvent):
+        private_key = self._get_private_key_from_workload()
+        csr = self._get_csr_from_workload()
+        if not private_key or csr != event.certificate_signing_request:
+            return
```

Comment:
We should at least log when things don't go as expected

Reply:
In this case, not doing anything when the event isn't meant for our unit is how we expect the unit to behave. 

I could still log this, but the result would be 5 lines of logs each time a new cert is generated from all units. If we have 5 cert requests for 5 peers for example the logs would look like:
```
logger.info("New certificate saved for unit 1")
logger.info("Cert not meant for unit 2. Skipping")
logger.info("Cert not meant for unit 3. Skipping")
logger.info("Cert not meant for unit 4. Skipping")
logger.info("Cert not meant for unit 5. Skipping")

logger.info("Cert not meant for unit 1. Skipping")
logger.info("New certificate saved for unit 2")
logger.info("Cert not meant for unit 3. Skipping")
logger.info("Cert not meant for unit 4. Skipping")
logger.info("Cert not meant for unit 5. Skipping")

logger.info("Cert not meant for unit 1. Skipping")
logger.info("Cert not meant for unit 2. Skipping")
logger.info("New certificate saved for unit 3")
logger.info("Cert not meant for unit 4. Skipping")
logger.info("Cert not meant for unit 5. Skipping")

logger.info("Cert not meant for unit 1. Skipping")
logger.info("Cert not meant for unit 2. Skipping")
logger.info("Cert not meant for unit 3. Skipping")
logger.info("New certificate saved for unit 4")
logger.info("Cert not meant for unit 5. Skipping")

logger.info("Cert not meant for unit 1. Skipping")
logger.info("Cert not meant for unit 2. Skipping")
logger.info("Cert not meant for unit 3. Skipping")
logger.info("Cert not meant for unit 4. Skipping")
logger.info("New certificate saved for unit 5")
```